src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -504,9 +504,10 @@ parseNullBinder = underscore *> return NullBinder
 
 parseIdentifierAndBinder :: TokenParser (String, Binder)
 parseIdentifierAndBinder =
-    do name <- lname <|> stringLiteral
+    do name <- lname
        b <- P.option (VarBinder (Ident name)) rest
        return (name, b)
+    <|> (,) <$> stringLiteral <*> rest
```

Comment:
You can have `{ foo: bar }` or `{ foo }` or `{ "!!!": baz }, but not `{ "!!!" }`, since that wouldn't be a valid `Ident`.